joplin/base/models/__init__.py

```diff
@@ -0,0 +1,165 @@
+        PageChooserPanel('topiccollection'),
+    ]
+
+    def __str__(self):
+        return self.topiccollection.text
+
+@register_snippet
+class ThreeOneOne(ClusterableModel):
+    title = models.CharField(max_length=DEFAULT_MAX_LENGTH)
+    url = models.URLField()
+
+    def __str__(self):
+        return self.title
+
+
+class ProcessPage(JanisPage):
+    janis_url_page_type = "processes"
+
+    department = models.ForeignKey(
+        'base.DepartmentPage',
+        on_delete=models.PROTECT,
+        verbose_name='Select a Department',
+        blank=True,
+        null=True,
+    )
+
+    description = models.TextField(blank=True)
+    image = models.ForeignKey(TranslatedImage, null=True, blank=True, on_delete=models.SET_NULL, related_name='+')
+    # TODO: Add images array field
+
+    base_form_class = custom_forms.ProcessPageForm
+
+    content_panels = [
+        InlinePanel('topics', label='Topics'),
+        FieldPanel('department'),
+        FieldPanel('description'),
+        ImageChooserPanel('image'),
+        InlinePanel('contacts', label='Contacts'),
+        InlinePanel('process_steps', label="Process steps"),
+    ]
+
+class ProcessPageStep(Orderable):
+    page = ParentalKey(ProcessPage, related_name='process_steps')
+    title = models.CharField(max_length=DEFAULT_MAX_LENGTH)
+    short_title = models.CharField(max_length=DEFAULT_MAX_LENGTH)
+    link_title = models.CharField(max_length=DEFAULT_MAX_LENGTH)
+    description = models.TextField(blank=True)
+    image = models.ForeignKey(TranslatedImage, null=True, blank=True, on_delete=models.SET_NULL, related_name='+')
+    overview_steps = RichTextField(features=WYSIWYG_GENERAL, verbose_name='Write out the steps a resident needs to take to use the service', blank=True)
+    detailed_content = RichTextField(features=WYSIWYG_GENERAL, verbose_name='Write any detailed content describing the process', blank=True)
+    quote = models.TextField(blank=True)
+
+    panels = [
+        FieldPanel('title'),
+        FieldPanel('short_title'),
+        FieldPanel('link_title'),
+        FieldPanel('description'),
+        ImageChooserPanel('image'),
+        FieldPanel('overview_steps'),
+        FieldPanel('detailed_content'),
+        FieldPanel('quote'),
+    ]
+
+class ProcessPageContact(ClusterableModel):
+    process = ParentalKey(ProcessPage, related_name='contacts')
+    contact = models.ForeignKey(Contact, related_name='+', on_delete=models.CASCADE)
+
+    panels = [
+        SnippetChooserPanel('contact'),
+    ]
+
+    def __str__(self):
+        return self.contact.name
+
+class ProcessPageTopic(ClusterableModel):
+    page = ParentalKey(ProcessPage, related_name='topics')
+    topic = models.ForeignKey('base.TopicPage',  verbose_name='Select a Topic', related_name='+', on_delete=models.CASCADE)
+
+    panels = [
+        PageChooserPanel('topic'),
+    ]
+
+    def __str__(self):
+        return self.topic.text
+
+@register_snippet
+class Department(ClusterableModel):
```

Comment:
pretty sure we've completely replaced this with DepartmentPage now, need to do some diving to be sure